components/modals/FilterModal.tsx

```diff
@@ -1,15 +1,270 @@
-import React from 'react';
-import { View } from 'react-native';
-
-// TODO: filter modal
-/**
- * -  should allow user to set level, topic, rooms, and session type filters
- * -  clicking the filter button should close the modal and set the filters on the sessions page
- * -  clicking cancel should close the modal
- */
-
-const FilterModal = () => {
-  return <View />;
+import { AntDesign } from '@expo/vector-icons';
+import { useTheme } from '@react-navigation/native';
+import React, { useState } from 'react';
+import { Modal, StyleSheet, TouchableOpacity, View } from 'react-native';
+import { SafeAreaView } from 'react-native-safe-area-context';
+import { typography } from '../../config/typography';
+import FilterButton from '../buttons/FilterButton';
+import PrimaryButton from '../buttons/PrimaryButton';
+import Row from '../common/Row';
+import Space from '../common/Space';
+import StyledText from '../common/StyledText';
+
+type Filters = {
+  level: string;
+  topic: string;
+  room: string;
+  sessionType: string;
+};
+
+type FilterModalProps = {
+  visible: boolean;
+  onClose: () => void;
+  onFilter: (filters: Filters) => void;
 };
 
+const FilterModal = ({ visible, onClose, onFilter }: FilterModalProps) => {
+  const { colors } = useTheme();
+  const { primary } = typography;
+
+  const [selectedLevel, setSelectedLevel] = useState<string>('Intermediate');
+  const [selectedTopic, setSelectedTopic] = useState<string>('UI/UX Design');
+  const [selectedRoom, setSelectedRoom] = useState<string>('Room 1');
+  const [selectedSessionType, setSelectedSessionType] = useState<string>('Session');
```

Comment:
@kharioki The UX of these filters seems off to me. I should be able to be granular in my selection, right? Imagine I want to attend "Intermediate" and "Advanced" sessions covering "Backend" and "APIs" topics across all the rooms. It is not possible to make such a filter currently.

Reply:
thats a good suggestion, I think I can update it to allow for multiple selection. lemme do that

Reply:
I updated this, now filters is an object of array items. cc @brianwachira you may want to consider this when implementing filter

<img width="955" alt="Screenshot 2023-07-14 at 01 08 44" src="https://github.com/droidconKE/droidconKE2023ReactNative/assets/22290070/9085b120-9df5-43e7-804d-7c1835a81da9">